gnat2goto/driver/tree_walk.adb

```diff
@@ -4095,6 +4113,61 @@ package body Tree_Walk is
       return  Ret;
    end Do_Signed_Integer_Definition;
 
+   ----------------------------------
+   -- Do_Floating_Point_Definition --
+   ----------------------------------
+
+   function Do_Floating_Point_Definition (N : Node_Id) return Irep is
+      E : constant Entity_Id := Defining_Entity (Parent (N));
+
+      --  This determines if ranges were specified or not
+      Scalar_Range_Ent : constant Node_Id := Scalar_Range (E);
+      Ret : constant Irep :=
+        New_Irep (if Nkind (Scalar_Range_Ent) = N_Real_Range_Specification
+                  then I_Bounded_Floatbv_Type
+                  else I_Floatbv_Type);
+      Esize_Width : constant Nat := UI_To_Int (Esize (E));
+      Digits_Expr : constant Node_Id := Digits_Expression (N);
+      Digits_Count : constant Uint := Intval (Digits_Expr);
+   begin
+      --  The digit precision is not used yet: this test is here only to show
+      --  how to access it.
+      if Digits_Count > 100 then
+         Report_Unhandled_Node_Empty (N, "Do_Floating_Point_Definition",
+                                      "This precision cannot be represented");
+         return New_Irep (I_Bounded_Floatbv_Type);
+      end if;
+
+      Set_Width (Ret, Integer (Esize_Width));
+      Set_F (Ret,
+             (case Esize_Width is
+                 when 32     => 23,
+                 --  23-bit mantissa, 8-bit exponent
+
+                 when 64     => 52,
+                 --  52-bit mantissa, 11-bit exponent
+
+                 when others => raise Program_Error));
```

Comment:
Should have a proper error message rather than a `Program_Error`?

Reply:
Done.